command/state_meta.go

```diff
@@ -103,24 +103,28 @@ func (c *StateMeta) lookupResourceInstanceAddr(state *states.State, allowMissing
 	case addrs.ModuleInstance:
 		// Matches all instances within the indicated module and all of its
 		// descendent modules.
+		found := false
```

Comment:
I think the logic around `found` is subtle enough that it might warrant a comment explaining that this is here to cope with the case of the empty intermediate module.